lib/collection.js

```diff
@@ -596,23 +607,40 @@ Collection.prototype.insertMany = function(docs, options, callback) {
  * @param {object} [options] Optional settings.
  * @param {(number|string)} [options.w] The write concern.
  * @param {number} [options.wtimeout] The write concern timeout.
- * @param {boolean} [options.j=false] Specify a journal write concern.
+ * @param {boolean} [options.j] Specify a journal write concern.
  * @param {boolean} [options.serializeFunctions=false] Serialize functions on any object.
  * @param {boolean} [options.ordered=true] Execute write operation in ordered or unordered fashion.
- * @param {boolean} [options.bypassDocumentValidation=false] Allow driver to bypass schema validation in MongoDB 3.2 or higher.
+ * @param {boolean} [options.bypassDocumentValidation] Allow driver to bypass schema validation in MongoDB 3.2 or higher.
  * @param {ClientSession} [options.session] optional session to use for this operation
  * @param {Collection~bulkWriteOpCallback} [callback] The command result callback
  * @return {Promise} returns Promise if no callback passed
  */
 Collection.prototype.bulkWrite = function(operations, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
-  options = options || { ordered: true };
 
   if (!Array.isArray(operations)) {
     throw MongoError.create({ message: 'operations must be an array of documents', driver: true });
   }
 
-  return executeOperation(this.s.topology, bulkWrite, [this, operations, options, callback]);
+  const bulkWriteSchema = {
+    w: { type: ['number', 'string'] },
+    wtimeout: { type: 'number' },
+    j: { type: 'boolean' },
+    serializeFunctions: { type: 'boolean' },
+    ordered: { type: 'boolean', default: true },
+    bypassDocumentValidation: { type: 'boolean' },
+    ignoreUndefined: { rejectUser: true },
```

Comment:
Let's all bikeshed a better name for this, I think `rejectUser` isn't very clear. Maybe `overrideOnly`, `ignore<some other word>`